### PR TITLE
feat(kubernetes): add new backend endpoints to api interface

### PR DESCRIPTION
**Pull request by mclarke47**
Signed-off-by: Matthew Clarke <mclarke@spotify.com>

Add the new backend endpoints to the client

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by mclarke47** [comment]
@benjdlambert if you get a sec!

**Review by jhaals — CHANGES_REQUESTED** [review]
Looks good overall, small suggestion regarding the API for evolvability. 😄

File: plugins/kubernetes/src/api/types.ts [hunk 1]
```diff
@@ -35,4 +38,13 @@ export interface KubernetesApi {
       oidcTokenProvider?: string | undefined;
     }[]
   >;
+  getWorkloadsByEntity(
+    auth: KubernetesRequestAuth,
+    entity: Entity,
+  ): Promise<ObjectsByEntityResponse>;
+  getCustomObjectsByEntity(
```
[the remark this answers is not in the document]
Comment (mclarke47) [reply plugins/kubernetes/src/api/types.ts:43]: done!

**Review by jhaals — APPROVED** [review]
Looking good, thanks!

**Pull request merged**